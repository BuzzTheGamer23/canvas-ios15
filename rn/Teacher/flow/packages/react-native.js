// @flow
export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}
