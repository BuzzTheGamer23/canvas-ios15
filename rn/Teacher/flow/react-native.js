//
// Copyright (C) 2017-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow
/* eslint-disable */

// define basic style prop types
// the internal ones are the best we have
import {
  type ____ViewStyleProp_Internal,
  type ____TextStyleProp_Internal,
  type ____ImageStyleProp_Internal,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes'

declare type ViewStyleProp = ____ViewStyleProp_Internal
declare type TextStyleProp = ____TextStyleProp_Internal
declare type ImageStyleProp = ____ImageStyleProp_Internal
