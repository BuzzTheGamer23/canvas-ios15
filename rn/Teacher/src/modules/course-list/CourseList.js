import React, { Component } from 'react'
import {
  Dimensions,
} from 'react-native'
import i18n from 'format-message'

import CourseCard from './components/course-card/CourseCard'
import GridView from '../../common/components/grid-view/GridView'

const { width: deviceWidth } = Dimensions.get('window')

type State = {
  width: number,
  padding: number,
  numItems: number,
}

type Props = {
  width: number,
}

const PADDING_CHANGE_WIDTH = 450
const MAX_CARD_WIDTH = 310

export default class CourseList extends Component<any, Props, State> {
  constructor (props: Props) {
    super(props)

    let width = props.width || deviceWidth

    this.state = {
      width: width,
      ...this.determineLayout(width),
    }
  }

  static navigatorButtons = {
    rightButtons: [
      {
        title: i18n({
          default: 'Edit',
          description: 'Shown at the top of the app to allow the user to edit their course list',
        }),
        id: 'edit',
        testID: 'e2e_rules',
      },
    ],
    leftButtons: [{
      title: i18n({
        default: 'Leave Feedback',
        description: 'Shown at the top of the app to allow the user to leave feedback',
      }),
      icon: require('../../images/feedback.png'),
    }],
  }

  determineLayout (width: number): {padding: number, numItems: number} {
    let newPadding = width > PADDING_CHANGE_WIDTH ? 12 : 8
    let newNumItems = Math.ceil(width / (MAX_CARD_WIDTH + newPadding * 2))
    return {
      padding: newPadding,
      numItems: newNumItems,
    }
  }

  onLayout = (event: any) => {
    if (event.nativeEvent.layout.width !== this.state.width) {
      this.setState(this.determineLayout(event.nativeEvent.layout.width))
    }
  }

  selectCourse (course: any) {
    this.props.navigator.push({
      screen: 'teacher.CourseDetails',
      passProps: { course },
    })
  }

  render (): React.Element<GridView> {
    let courses = [{
      color: '#27B9CD',
      image_download_url: 'https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg',
      name: 'Biology 101',
      course_code: 'BIO 101',
      id: 1,
    }, {
      color: '#8F3E97',
      image_download_url: 'https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg',
      name: 'American Literature Psysicks foobar hello world 401',
      course_code: 'LIT 401',
      id: 2,
    }, {
      color: '#8F3E97',
      image_download_url: 'https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg',
      name: 'Foobar 102',
      course_code: 'FOO 102',
      id: 3,
    }]

    let cardStyles = {
      flex: 1,
      margin: this.state.padding,
    }

    return (
      <GridView
        onLayout={this.onLayout}
        style={{ padding: this.state.padding }}
        placeholderStyle={cardStyles}
        data={courses}
        itemsPerRow={this.state.numItems}
        renderItem={(rowData: Course) =>
          <CourseCard
            style={cardStyles}
            course={rowData}
            key={rowData.course_code}
            onPress={() => this.selectCourse(rowData)}
          />
        }
      />
    )
  }
}
