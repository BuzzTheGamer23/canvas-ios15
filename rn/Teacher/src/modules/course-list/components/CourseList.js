// @flow

import React, { Component, PropTypes } from 'react'
import { View } from 'react-native'
import CourseCard from './CourseCard'
import GridView from '../../../common/components/grid-view/GridView'

type State = {
  width: number,
  padding: number,
  numItems: number,
}

type Props = {
  width: number,
  selectCourse: (course: Course) => void,
  header?: React.Element<*>,
  customColors: { [string]: string },
  courses: Array<Course>,
  error?: string,
  pending: number,
}

const PADDING_CHANGE_WIDTH = 450
const MAX_CARD_WIDTH = 310

export default class CourseList extends Component {
  props: Props
  state: State

  constructor (props: Props) {
    super(props)

    this.state = {
      width: props.width,
      ...this.determineLayout(props.width),
    }
  }

  determineLayout (width: number): {padding: number, numItems: number} {
    let newPadding = width > PADDING_CHANGE_WIDTH ? 12 : 8
    let newNumItems = Math.ceil(width / (MAX_CARD_WIDTH + newPadding * 2))
    return {
      padding: newPadding,
      numItems: newNumItems,
    }
  }

  onLayout = (event: any) => {
    if (event.nativeEvent.layout.width !== this.state.width) {
      this.setState(this.determineLayout(event.nativeEvent.layout.width))
    }
  }

  selectCourse (course: any) {
    this.props.selectCourse(course)
  }

  renderHeader = () => {
    return (
      <View style={{ paddingHorizontal: this.state.padding }}>
        {this.props.header}
      </View>
    )
  }

  render (): React.Element<*> {
    let cardStyles = {
      flex: 1,
      margin: this.state.padding,
    }

    return (
      <GridView
        onLayout={this.onLayout}
        style={{ padding: this.state.padding }}
        placeholderStyle={cardStyles}
        data={this.props.courses}
        itemsPerRow={this.state.numItems}
        renderItem={(rowData: Course) =>
          <CourseCard
            style={cardStyles}
            course={rowData}
            color={this.props.customColors[rowData.id.toString()]}
            key={rowData.course_code}
            onPress={() => this.selectCourse(rowData)}
          />
        }
        renderHeader={this.renderHeader}
      />
    )
  }
}

const courseListShape = PropTypes.shape({
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  course_code: PropTypes.string.isRequired,
  short_name: PropTypes.string,
  color: PropTypes.string,
  image_download_url: PropTypes.string,
}).isRequired

CourseList.propTypes = {
  courses: PropTypes.arrayOf(courseListShape).isRequired,
  pending: PropTypes.number,
  error: PropTypes.string,
}
