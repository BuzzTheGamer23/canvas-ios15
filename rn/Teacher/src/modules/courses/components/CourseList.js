//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View } from 'react-native'
import CourseCard from './CourseCard'
import GridView from '../../../common/components/GridView'
import type { CourseProps } from '../course-prop-types'

type State = {
  width: number,
  padding: number,
  numItems: number,
}

type Props = {
  width: number,
  selectCourse: (course: Course) => void,
  header?: React.Element<*>,
  courses: Array<CourseProps>,
  error?: string,
  pending: number,
  onCoursePreferencesPressed: (courseId: string) => void,
  onRefresh: Function,
}

const PADDING_CHANGE_WIDTH = 450
const MAX_CARD_WIDTH = 295

export default class CourseList extends Component {
  props: Props
  state: State

  constructor (props: Props) {
    super(props)

    this.state = {
      width: props.width,
      ...this.determineLayout(props.width),
    }
  }

  determineLayout (width: number): {padding: number, numItems: number} {
    let newPadding = width > PADDING_CHANGE_WIDTH ? 12 : 8
    let newNumItems = Math.ceil(width / (MAX_CARD_WIDTH + newPadding * 2))
    return {
      padding: newPadding,
      numItems: newNumItems,
    }
  }

  onLayout = (event: any) => {
    if (event.nativeEvent.layout.width !== this.state.width) {
      this.setState(this.determineLayout(event.nativeEvent.layout.width))
    }
  }

  renderHeader = () => {
    return (
      <View style={{ paddingHorizontal: this.state.padding, marginTop: this.state.padding }}>
        {this.props.header}
      </View>
    )
  }

  render () {
    let cardStyles = {
      flex: 1,
      margin: this.state.padding,
    }

    return (
      <GridView
        onLayout={this.onLayout}
        showsVerticalScrollIndicator={false}
        style={{ marginHorizontal: this.state.padding, overflow: 'visible' }}
        placeholderStyle={cardStyles}
        data={this.props.courses}
        itemsPerRow={this.state.numItems}
        renderItem={(rowData: CourseProps) =>
          <CourseCard
            style={cardStyles}
            course={rowData}
            color={rowData.color}
            key={`${rowData.course_code}_${rowData.id}`}
            onPress={this.props.selectCourse}
            onCoursePreferencesPressed={this.props.onCoursePreferencesPressed}
          />
        }
        renderHeader={this.renderHeader}
        onRefresh={this.props.onRefresh}
      />
    )
  }
}

const coursePropsShape = PropTypes.shape({
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  course_code: PropTypes.string.isRequired,
  short_name: PropTypes.string,
  color: PropTypes.string,
  image_download_url: PropTypes.string,
})

CourseList.propTypes = {
  courses: PropTypes.arrayOf(coursePropsShape).isRequired,
  pending: PropTypes.number,
  error: PropTypes.string,
}
