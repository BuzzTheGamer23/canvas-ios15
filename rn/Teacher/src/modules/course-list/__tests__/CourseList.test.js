/* @flow */

import 'react-native'
import React from 'react'
import { CourseList } from '../CourseList.js'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

const courses: Course[] = [{
  color: '#27B9CD',
  image_download_url: 'https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg',
  name: 'Biology 101',
  course_code: 'BIO 101',
  short_name: 'BIO 101',
  id: 1,
}, {
  color: '#8F3E97',
  image_download_url: 'https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg',
  name: 'American Literature Psysicks foobar hello world 401',
  course_code: 'LIT 401',
  short_name: 'LIT 401',
  id: 2,
}, {
  color: '#8F3E97',
  image_download_url: 'https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg',
  name: 'Foobar 102',
  course_code: 'FOO 102',
  id: 3,
  short_name: 'FOO 102',
}]

let refreshCourses = () => {}

test('renders correctly', () => {
  let tree = renderer.create(
    <CourseList width={320} refreshCourses={ refreshCourses } courses={ courses } />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders correctly with wide device', () => {
  let tree = renderer.create(
    <CourseList width={768} refreshCourses={ refreshCourses } courses={ courses } />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('refreshCourses was called', () => {
  let didCallRefresh = false
  let refresh = () => {
    didCallRefresh = true
  }
  let tree = renderer.create(
    <CourseList width={320} refreshCourses={ refresh } courses={ courses }/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
  expect(didCallRefresh).toBe(true)
})
