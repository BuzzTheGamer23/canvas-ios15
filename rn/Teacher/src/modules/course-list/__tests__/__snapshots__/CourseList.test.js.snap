// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "padding": 8,
      },
    ]
  }
>
  <View>
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-around",
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Array [
                  Object {
                    "borderColor": "#e3e3e3",
                    "borderWidth": 0.5,
                    "height": 160,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": 0,
                    },
                    "shadowOpacity": 0.2,
                    "shadowRadius": 1,
                  },
                  Object {
                    "flex": 1,
                    "margin": 8,
                  },
                ],
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 2,
                    "justifyContent": "space-between",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#27B9CD",
                      "height": 80,
                    }
                  }
                />
                <Image
                  source={
                    Object {
                      "uri": "https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg",
                    }
                  }
                  style={
                    Object {
                      "height": 80,
                      "opacity": 0.2,
                      "top": -80,
                    }
                  }
                />
                <Image
                  source={1}
                  style={
                    Object {
                      "position": "absolute",
                      "right": 5,
                      "top": 10,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "flex": 2,
                    "marginHorizontal": 10,
                    "marginTop": 20,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={2}
                  style={
                    Object {
                      "color": "#27B9CD",
                      "fontSize": 14,
                      "fontWeight": "500",
                    }
                  }
                >
                  Biology 101
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#8B969D",
                      "fontSize": 11,
                      "fontWeight": "500",
                    }
                  }
                >
                  BIO 101
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-around",
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Array [
                  Object {
                    "borderColor": "#e3e3e3",
                    "borderWidth": 0.5,
                    "height": 160,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": 0,
                    },
                    "shadowOpacity": 0.2,
                    "shadowRadius": 1,
                  },
                  Object {
                    "flex": 1,
                    "margin": 8,
                  },
                ],
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 2,
                    "justifyContent": "space-between",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#8F3E97",
                      "height": 80,
                    }
                  }
                />
                <Image
                  source={
                    Object {
                      "uri": "https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg",
                    }
                  }
                  style={
                    Object {
                      "height": 80,
                      "opacity": 0.2,
                      "top": -80,
                    }
                  }
                />
                <Image
                  source={1}
                  style={
                    Object {
                      "position": "absolute",
                      "right": 5,
                      "top": 10,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "flex": 2,
                    "marginHorizontal": 10,
                    "marginTop": 20,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={2}
                  style={
                    Object {
                      "color": "#8F3E97",
                      "fontSize": 14,
                      "fontWeight": "500",
                    }
                  }
                >
                  American Literature Psysicks foobar hello world 401
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#8B969D",
                      "fontSize": 11,
                      "fontWeight": "500",
                    }
                  }
                >
                  LIT 401
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-around",
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Array [
                  Object {
                    "borderColor": "#e3e3e3",
                    "borderWidth": 0.5,
                    "height": 160,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": 0,
                    },
                    "shadowOpacity": 0.2,
                    "shadowRadius": 1,
                  },
                  Object {
                    "flex": 1,
                    "margin": 8,
                  },
                ],
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 2,
                    "justifyContent": "space-between",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#8F3E97",
                      "height": 80,
                    }
                  }
                />
                <Image
                  source={
                    Object {
                      "uri": "https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg",
                    }
                  }
                  style={
                    Object {
                      "height": 80,
                      "opacity": 0.2,
                      "top": -80,
                    }
                  }
                />
                <Image
                  source={1}
                  style={
                    Object {
                      "position": "absolute",
                      "right": 5,
                      "top": 10,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "flex": 2,
                    "marginHorizontal": 10,
                    "marginTop": 20,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={2}
                  style={
                    Object {
                      "color": "#8F3E97",
                      "fontSize": 14,
                      "fontWeight": "500",
                    }
                  }
                >
                  Foobar 102
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#8B969D",
                      "fontSize": 11,
                      "fontWeight": "500",
                    }
                  }
                >
                  FOO 102
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`renders correctly with wide device 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "padding": 12,
      },
    ]
  }
>
  <View>
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-around",
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Array [
                  Object {
                    "borderColor": "#e3e3e3",
                    "borderWidth": 0.5,
                    "height": 160,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": 0,
                    },
                    "shadowOpacity": 0.2,
                    "shadowRadius": 1,
                  },
                  Object {
                    "flex": 1,
                    "margin": 12,
                  },
                ],
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 2,
                    "justifyContent": "space-between",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#27B9CD",
                      "height": 80,
                    }
                  }
                />
                <Image
                  source={
                    Object {
                      "uri": "https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg",
                    }
                  }
                  style={
                    Object {
                      "height": 80,
                      "opacity": 0.2,
                      "top": -80,
                    }
                  }
                />
                <Image
                  source={1}
                  style={
                    Object {
                      "position": "absolute",
                      "right": 5,
                      "top": 10,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "flex": 2,
                    "marginHorizontal": 10,
                    "marginTop": 20,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={2}
                  style={
                    Object {
                      "color": "#27B9CD",
                      "fontSize": 14,
                      "fontWeight": "500",
                    }
                  }
                >
                  Biology 101
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#8B969D",
                      "fontSize": 11,
                      "fontWeight": "500",
                    }
                  }
                >
                  BIO 101
                </Text>
              </View>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Array [
                  Object {
                    "borderColor": "#e3e3e3",
                    "borderWidth": 0.5,
                    "height": 160,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": 0,
                    },
                    "shadowOpacity": 0.2,
                    "shadowRadius": 1,
                  },
                  Object {
                    "flex": 1,
                    "margin": 12,
                  },
                ],
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 2,
                    "justifyContent": "space-between",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#8F3E97",
                      "height": 80,
                    }
                  }
                />
                <Image
                  source={
                    Object {
                      "uri": "https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg",
                    }
                  }
                  style={
                    Object {
                      "height": 80,
                      "opacity": 0.2,
                      "top": -80,
                    }
                  }
                />
                <Image
                  source={1}
                  style={
                    Object {
                      "position": "absolute",
                      "right": 5,
                      "top": 10,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "flex": 2,
                    "marginHorizontal": 10,
                    "marginTop": 20,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={2}
                  style={
                    Object {
                      "color": "#8F3E97",
                      "fontSize": 14,
                      "fontWeight": "500",
                    }
                  }
                >
                  American Literature Psysicks foobar hello world 401
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#8B969D",
                      "fontSize": 11,
                      "fontWeight": "500",
                    }
                  }
                >
                  LIT 401
                </Text>
              </View>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Array [
                  Object {
                    "borderColor": "#e3e3e3",
                    "borderWidth": 0.5,
                    "height": 160,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": 0,
                    },
                    "shadowOpacity": 0.2,
                    "shadowRadius": 1,
                  },
                  Object {
                    "flex": 1,
                    "margin": 12,
                  },
                ],
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 2,
                    "justifyContent": "space-between",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#8F3E97",
                      "height": 80,
                    }
                  }
                />
                <Image
                  source={
                    Object {
                      "uri": "https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg",
                    }
                  }
                  style={
                    Object {
                      "height": 80,
                      "opacity": 0.2,
                      "top": -80,
                    }
                  }
                />
                <Image
                  source={1}
                  style={
                    Object {
                      "position": "absolute",
                      "right": 5,
                      "top": 10,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "flex": 2,
                    "marginHorizontal": 10,
                    "marginTop": 20,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={2}
                  style={
                    Object {
                      "color": "#8F3E97",
                      "fontSize": 14,
                      "fontWeight": "500",
                    }
                  }
                >
                  Foobar 102
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#8B969D",
                      "fontSize": 11,
                      "fontWeight": "500",
                    }
                  }
                >
                  FOO 102
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
